### PR TITLE
Upgrade solidity-coverage to ^0.7.0

### PR DESCRIPTION
**Pull request by cgewecke**
#436 

The new solidity-coverage installs as a truffle-plugin. Leaving a few notes in the diff to explain the installation since it's slightly outside the norm.

### CODE REVIEW

File: .travis.yml
Comment (cgewecke) [inline .travis.yml:12]: Coverage now uses your `development` network settings, but needs to launch the ganache server itself because it hooks directly into the evm to track code execution. 

This line had to be removed to avoid a 'port already in use' error.

File: package.json
Comment (cgewecke) [inline package.json]:
```
--temp build
```
means, "write build artifacts to the `build` folder." In SC 0.7.0 these are normally written to a temporary folder called '.coverage_artifacts' and thrown away. 

This flag needs to be set because your scripts rely on artifacts being available via the `build` path here:
https://github.com/gnosis/dex-contracts/blob/5ab2c4e231c96621fad6fada12100de2b97fdda6/src/index.js#L9-L13

NB: The `build` folder **gets deleted** after the coverage command is run (relevant if you're sometimes storing real network deployment data there).

File: src/migration/PoC_dfusion.js
Comment (cgewecke) [inline src/migration/PoC_dfusion.js:36]: Without first calling `.deployed()` on libraries which are linked to other contracts about to be deployed, I see  deployments fail with this error:
```
Error: 
    at Object.link (/Users/cgewecke/code/sc-forks/dex-contracts/node_modules/truffle/build/webpack:/packages/deployer/src/linker.js:29:1)
    at process._tickCallback (internal/process/next_tick.js:68:7)
Truffle v5.1.6 (core: 5.1.6)
Node v10.15.3
```
Honestly don't know why this is necessary or why it's surfacing here in the plugin version of the tool. Feels like a bug within Truffle migrations - perhaps the artifact object has properties managed by getters/setters that aren't populated correctly unless `get` is called first in some contexts.  Idk.

It may also be specific to artifacts stored in node_modules - haven't seen it before and will keep an eye out.

File: .travis.yml
Comment (bh2smith) [reply .travis.yml:12]: This is very nice of you to assist with these fixes! Thanks!

**Review by bh2smith — APPROVED** [review]
Very cool @cgewecke! Thanks for your detailed explanations and contribution.  This looks great to me, although we will wait on one more review before merge!

File: package.json
Comment (fleupold) [reply package.json]: I wonder if it would be better to build the artefacts explicitly (so that they persist) and having coverage use the throw_away folder. Otherwise now we would delete artefacts, even if we `truffle compiled` them before, right?

**Review by fleupold — APPROVED** [review]
Looks good. Maybe we should not delete the build folder after the coverage run though and rather create a persistent version by running truffle compile explicitly.

File: package.json
Comment (cgewecke) [reply package.json]: @fleupold 👍 

In c77baee531755c31304d3e48acdaeb93cea90f61 have removed the `--temp build` flag after modifying a `require` path in test/utilities to load encoding.js directly from its file instead of from `src/index`.  

That looks like the only place Node wasn't able to resolve paths correctly when using an alternative location for artifacts.

The build folder won't be touched or deleted by coverage anymore.

**Comment by fleupold** [comment]
Awesome! Are we good to merge then?

**Review by nlordell — APPROVED** [review]

**Pull request merged**
